apis/supervisor/config/v1alpha1/types_oidcclient.go.tmpl

```diff
@@ -71,6 +71,28 @@ type OIDCClientSpec struct {
 	// +listType=set
 	// +kubebuilder:validation:MinItems=1
 	AllowedScopes []Scope `json:"allowedScopes"`
+
+	// tokenLifetimes are the optional overrides of token lifetimes for an OIDCClient.
+	// +optional
+	TokenLifetimes OIDCClientTokenLifetimes `json:"tokenLifetimes,omitempty"`
+}
+
+// OIDCClientTokenLifetimes describes the optional overrides of token lifetimes for an OIDCClient.
+type OIDCClientTokenLifetimes struct {
+	// idTokenSeconds is the lifetime of ID tokens issued to this client, in seconds. This will choose the lifetime of
+	// ID tokens returned by the authorization flow and the refresh grant. It will not influence the lifetime of the ID
+	// tokens returned by RFC8693 token exchange. When null, a short-lived default value of approximately several
+	// minutes will be used. This value must be between 30 and 1,800 seconds (30 minutes), inclusive. It is recommended
+	// to make these tokens short-lived to force the client to perform the refresh grant often, because the refresh
+	// grant will check with the external identity provider to decide if it is acceptable for the end user to continue
+	// their session, and will refresh the end user's group memberships in case they have changed in the external
+	// identity provider. Giving these tokens a long life will allow the end user to continue to use a token while
+	// avoiding these updates from the external identity provider, which is generally not desirable from a security
+	// point of view.
```

Comment:
It's not super clear from the grammar what `generally not desirable from a security point of view` refers to. Could we say it this way?

> Giving these tokens a long life is generally not desirable from a security point of view, since that will allow the end user to continue to use a token while avoiding these updates from the external identity provider.

Reply:
Updating as suggested.